config/config.go

```diff
@@ -297,9 +297,13 @@ func (cfg *Config) NewNode() (host.Host, error) {
 			ConnectionGater:    cfg.ConnectionGater,
 			Reporter:           cfg.Reporter,
 			PeerKey:            autonatPrivKey,
+		}
 
-			Peerstore: pstoremem.NewPeerstore(),
+		ps, err := pstoremem.NewPeerstore()
```

Comment:
Nit: construct it before the initialization of `autoNatCfg`, as we do for `autonatPrivKey`.